plugins/techdocs-backend/src/search/DefaultTechDocsDocumentGenerator.test.ts

```diff
@@ -0,0 +1,223 @@
+      name: 'test-entity-with-docs',
+      description: 'Documented description',
+      annotations: {
+        'backstage.io/techdocs-ref': './',
+      },
+    },
+    spec: {
+      type: 'dog',
+      lifecycle: 'experimental',
+      owner: 'someone',
+    },
+  },
+  {
+    apiVersion: 'backstage.io/v1alpha1',
+    kind: 'Component',
+    metadata: {
+      name: 'test-entity',
+      description: 'The expected description',
+    },
+    spec: {
+      type: 'some-type',
+      lifecycle: 'experimental',
+    },
+  },
+];
+
+describe('DefaultTechDocsCollator with legacyPathCasing configuration', () => {
+  let mockDiscoveryApi: jest.Mocked<PluginEndpointDiscovery>;
+  let collator: DefaultTechDocsDocumentGenerator;
+
+  const worker = setupServer();
+  msw.setupDefaultHandlers(worker);
+  beforeEach(() => {
+    mockDiscoveryApi = {
+      getBaseUrl: jest.fn().mockResolvedValue('http://test-backend'),
+      getExternalBaseUrl: jest.fn(),
+    };
+    const mockConfig = new ConfigReader({
+      techdocs: {
+        legacyUseCaseSensitiveTripletPaths: true,
+      },
+    });
+    collator = DefaultTechDocsDocumentGenerator.fromConfig(mockConfig, {
+      discovery: mockDiscoveryApi,
+      logger,
+      legacyPathCasing: true,
+    });
+
+    worker.use(
+      rest.get(
+        'http://test-backend/static/docs/default/Component/test-entity-with-docs/search/search_index.json',
+        (_, res, ctx) => res(ctx.status(200), ctx.json(mockSearchDocIndex)),
+      ),
+      rest.get('http://test-backend/entities', (_, res, ctx) =>
+        res(ctx.status(200), ctx.json(expectedEntities)),
+      ),
+    );
+  });
+
+  it('fetches from the configured catalog and tech docs services', async () => {
+    const collatorStream = Readable.from(collator.execute());
+    const { documents } = await TestPipeline.withSubject(
+      collatorStream,
+    ).execute();
+    expect(mockDiscoveryApi.getBaseUrl).toHaveBeenCalledWith('catalog');
+    expect(mockDiscoveryApi.getBaseUrl).toHaveBeenCalledWith('techdocs');
+    expect(documents).toHaveLength(mockSearchDocIndex.docs.length);
+  });
+
+  it('should create documents for each tech docs search index', async () => {
+    const collatorStream = Readable.from(collator.execute());
+    const { documents } = await TestPipeline.withSubject(
+      collatorStream,
+    ).execute();
+    const entity = expectedEntities[0];
+    documents.forEach((document, idx) => {
+      expect(document).toMatchObject({
+        title: mockSearchDocIndex.docs[idx].title,
+        location: `/docs/default/Component/${entity.metadata.name}/${mockSearchDocIndex.docs[idx].location}`,
+        text: mockSearchDocIndex.docs[idx].text,
+        namespace: 'default',
+        entityTitle: entity!.metadata.title,
+        componentType: entity!.spec!.type,
+        lifecycle: entity!.spec!.lifecycle,
+        owner: '',
+      });
+    });
+  });
+});
+
+describe('DefaultTechDocsCollator', () => {
```

Comment:
similar comment to above, it feels like this is testing the `DefaultTechDocsDocumentGenerator` and not `DefaultTechDocsCollator`